### PR TITLE
Add navbar

### PR DESCRIPTION
**Pull request by JustAGhost23**
Added a navbar, still need to decide how to conditionally not display navbar on login page.

### CODE REVIEW

**Review by soumitradev — CHANGES_REQUESTED** [review]
Few quick changes, the auth navbar layout thing will be done in a separate PR

File: frontend/src/components/navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  const createMutation = useMutation({
+    mutationFn: () => {
+      return axios.post<{ message: string; id: number }>(
+        "/api/timetable/create",
+      );
+    },
+    onSuccess: (response) => {
+      queryClient.invalidateQueries({ queryKey: ["user"] });
+      router.navigate({ to: `/edit/${response.data.id}` });
+    },
```
[the remark this answers is not in the document]
Comment (JustAGhost23) [reply frontend/src/components/navbar.tsx]: Done in [32f3669](https://github.com/crux-bphc/chronofactorem-rewrite/pull/126/commits/32f36696111eeb0a157c92cef3da4cdcac42450e)

File: frontend/src/components/navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+              <Link to="/login">
+                <LogOut className="mr-2 h-4 w-4" />
+                <span>Log out</span>
+              </Link>
+            </DropdownMenuItem>
+          </DropdownMenuContent>
+        </DropdownMenu>
+      </div>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (JustAGhost23) [reply frontend/src/components/navbar.tsx]: Done in [93a9b42](https://github.com/crux-bphc/chronofactorem-rewrite/pull/126/commits/93a9b429913f8ba2eb8a73eed212849d02f2cb18)

**Review by soumitradev — APPROVED** [review]
LGTM

**Pull request merged**